### ISSUE TITLE
`Bloch().add_points()` breaks in Matplotlib 3.1.0 (works in Matplotlib 3.0.3)

### ISSUE THREAD

**Issue by nathanshammah** [qutip/qutip#1027]
```
from qutip import *
import numpy as np
#from mayavi import mlab
import matplotlib.pyplot as plt

b = Bloch()
pnt = [1/np.sqrt(3),1/np.sqrt(3),1/np.sqrt(3)]
b.add_points(pnt)
Bloch().add_points(pnt)
b.show()
qutip.about()
```
gives the error in a Jupyter notebook:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/miniconda3/lib/python3.7/site-packages/IPython/core/formatters.py in __call__(self, obj)
    339                 pass
    340             else:
--> 341                 return printer(obj)
    342             # Finally look for special method names
    343             method = get_real_method(obj, self.print_method)

/miniconda3/lib/python3.7/site-packages/IPython/core/pylabtools.py in <lambda>(fig)
    242 
    243     if 'png' in formats:
--> 244         png_formatter.for_type(Figure, lambda fig: print_figure(fig, 'png', **kwargs))
    245     if 'retina' in formats or 'png2x' in formats:
    246         png_formatter.for_type(Figure, lambda fig: retina_figure(fig, **kwargs))

/miniconda3/lib/python3.7/site-packages/IPython/core/pylabtools.py in print_figure(fig, fmt, bbox_inches, **kwargs)
    126 
    127     bytes_io = BytesIO()
--> 128     fig.canvas.print_figure(bytes_io, **kw)
    129     data = bytes_io.getvalue()
    130     if fmt == 'svg':

/miniconda3/lib/python3.7/site-packages/matplotlib/backend_bases.py in print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, **kwargs)
   2054                         orientation=orientation,
   2055                         dryrun=True,
-> 2056                         **kwargs)
   2057                     renderer = self.figure._cachedRenderer
   2058                     bbox_artists = kwargs.pop("bbox_extra_artists", None)

/miniconda3/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py in print_png(self, filename_or_obj, metadata, pil_kwargs, *args, **kwargs)
    525 
    526         else:
--> 527             FigureCanvasAgg.draw(self)
    528             renderer = self.get_renderer()
    529             with cbook._setattr_cm(renderer, dpi=self.figure.dpi), \

/miniconda3/lib/python3.7/site-packages/matplotlib/backends/backend_agg.py in draw(self)
    386         self.renderer = self.get_renderer(cleared=True)
    387         with RendererAgg.lock:
--> 388             self.figure.draw(self.renderer)
    389             # A GUI class may be need to update a window using this draw, so
    390             # don't forget to call the superclass.

/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py in draw_wrapper(artist, renderer, *args, **kwargs)
     36                 renderer.start_filter()
     37 
---> 38             return draw(artist, renderer, *args, **kwargs)
     39         finally:
     40             if artist.get_agg_filter() is not None:

/miniconda3/lib/python3.7/site-packages/matplotlib/figure.py in draw(self, renderer)
   1707             self.patch.draw(renderer)
   1708             mimage._draw_list_compositing_images(
-> 1709                 renderer, self, artists, self.suppressComposite)
   1710 
   1711             renderer.close_group('figure')

/miniconda3/lib/python3.7/site-packages/matplotlib/image.py in _draw_list_compositing_images(renderer, parent, artists, suppress_composite)
    133     if not_composite or not has_images:
    134         for a in artists:
--> 135             a.draw(renderer)
    136     else:
    137         # Composite any adjacent images together

/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py in draw_wrapper(artist, renderer, *args, **kwargs)
     36                 renderer.start_filter()
     37 
---> 38             return draw(artist, renderer, *args, **kwargs)
     39         finally:
     40             if artist.get_agg_filter() is not None:

/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/axes3d.py in draw(self, renderer)
    290                 sorted(self.collections,
    291                        key=lambda col: col.do_3d_projection(renderer),
--> 292                        reverse=True)):
    293             col.zorder = zorder_offset + i
    294         for i, patch in enumerate(

/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/axes3d.py in <lambda>(col)
    289         for i, col in enumerate(
    290                 sorted(self.collections,
--> 291                        key=lambda col: col.do_3d_projection(renderer),
    292                        reverse=True)):
    293             col.zorder = zorder_offset + i

/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/art3d.py in do_3d_projection(self, renderer)
    543         self.set_facecolors(fcs)
    544 
--> 545         ecs = (_zalpha(self._edgecolor3d, vzs) if self._depthshade else
    546                self._edgecolor3d)
    547         ecs = mcolors.to_rgba_array(ecs, self._alpha)

/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/art3d.py in _zalpha(colors, zs)
    845     norm = Normalize(min(zs), max(zs))
    846     sats = 1 - norm(zs) * 0.7
--> 847     rgba = np.broadcast_to(mcolors.to_rgba_array(colors), (len(zs), 4))
    848     return np.column_stack([rgba[:, :3], rgba[:, 3] * sats])
    849 

/miniconda3/lib/python3.7/site-packages/numpy/lib/stride_tricks.py in broadcast_to(array, shape, subok)
    174            [1, 2, 3]])
    175     """
--> 176     return _broadcast_to(array, shape, subok=subok, readonly=True)
    177 
    178 

/miniconda3/lib/python3.7/site-packages/numpy/lib/stride_tricks.py in _broadcast_to(array, shape, subok, readonly)
    126     it = np.nditer(
    127         (array,), flags=['multi_index', 'refs_ok', 'zerosize_ok'] + extras,
--> 128         op_flags=[op_flag], itershape=shape, order='C')
    129     with it:
    130         # never really has writebackifcopy semantics

ValueError: operands could not be broadcast together with remapped shapes [original->remapped]: (0,4) and requested shape (2,4)

<Figure size 360x360 with 1 Axes>

QuTiP: Quantum Toolbox in Python
Copyright (c) 2011 and later.
A. J. Pitchford, P. D. Nation, R. J. Johansson, A. Grimsmo, and C. Granade

QuTiP Version:      4.3.1
Numpy Version:      1.15.4
Scipy Version:      1.2.1
Cython Version:     0.29.8
Matplotlib Version: 3.1.0
Python Version:     3.7.3
Number of CPUs:     2
BLAS Info:          OPENBLAS
OPENMP Installed:   False
INTEL MKL Ext:      False
Platform Info:      Darwin (x86_64)
Installation path:  /miniconda3/lib/python3.7/site-packages/qutip
==============================================================================
Please cite QuTiP in your publication.
==============================================================================
For your convenience a bibtex file can be easily generated using `qutip.cite()`
```
The error from a python script seems to shed more light into what is happening: 
```
/miniconda3/lib/python3.7/site-packages/qutip/bloch.py:631: MatplotlibDeprecationWarning: Passing the block parameter of show() positionally is deprecated since Matplotlib 3.1; the parameter will become keyword-only in 3.3.
  plt.show(self.fig)
Traceback (most recent call last):
  File "/miniconda3/lib/python3.7/site-packages/matplotlib/backends/backend_macosx.py", line 80, in _draw
    self.figure.draw(renderer)
  File "/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/miniconda3/lib/python3.7/site-packages/matplotlib/figure.py", line 1709, in draw
    renderer, self, artists, self.suppressComposite)
  File "/miniconda3/lib/python3.7/site-packages/matplotlib/image.py", line 135, in _draw_list_compositing_images
    a.draw(renderer)
  File "/miniconda3/lib/python3.7/site-packages/matplotlib/artist.py", line 38, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/axes3d.py", line 292, in draw
    reverse=True)):
  File "/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/axes3d.py", line 291, in <lambda>
    key=lambda col: col.do_3d_projection(renderer),
  File "/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/art3d.py", line 545, in do_3d_projection
    ecs = (_zalpha(self._edgecolor3d, vzs) if self._depthshade else
  File "/miniconda3/lib/python3.7/site-packages/mpl_toolkits/mplot3d/art3d.py", line 847, in _zalpha
    rgba = np.broadcast_to(mcolors.to_rgba_array(colors), (len(zs), 4))
  File "/miniconda3/lib/python3.7/site-packages/numpy/lib/stride_tricks.py", line 176, in broadcast_to
    return _broadcast_to(array, shape, subok=subok, readonly=True)
  File "/miniconda3/lib/python3.7/site-packages/numpy/lib/stride_tricks.py", line 128, in _broadcast_to
    op_flags=[op_flag], itershape=shape, order='C')
ValueError: operands could not be broadcast together with remapped shapes [original->remapped]: (0,4) and requested shape (2,4)
```
The same snippet works properly in Matplotlib 3.0.3, so it's something in Matplotlib 3.1.0.

**Comment by taalexander** [comment]
Can also confirm this is an issue for me with matplotlib 3.1.0

**Comment by pops2724** [comment]
I also have an issue when running this in Jupyter Notebook.

**Comment by XiaoTailong** [comment]
I have the same problem in Bloch() and show() functions in Mac OS 10.14. I cannot plot the Bloch sphere. It seems the problem of Matplotlib. But I am not sure.

**Comment by nathanshammah** [comment]
For the time being, a fix is to downgrade Matplotlib:

conda install matplotlib==3.0.3

**Comment by nathanshammah** [comment]
``b.add_points([1/np.sqrt(3),1/np.sqrt(3),1/np.sqrt(3)],meth="l")``
Gives no error, but produces no output. The other methods fail.

**Comment by chaithumcn3** [comment]
The above code is working fine with Matplotlib 3.1.3. But when downgrading it to 3.1.0 and running the code is causing error.

**Comment by MichKe95** [comment]
I can reproduce the above errors with Matplotlib 3.2.1.

**Comment by XiaoTailong** [comment]
Matlibplot 3.0.3may work will


发自我的iPhone

------------------ Original ------------------

**Comment by ZhiwenZong** [comment]
I have the same question

**Comment by hodgestar** [comment]
This was fixed by @nathanshammah in #1097 and we just forgot to close the issue.
